sdk/core/core-client-rest/src/clientHelpers.ts

```diff
@@ -71,7 +72,12 @@ export function createDefaultPipeline(
 ): Pipeline {
   const pipeline = createPipelineFromOptions(options);
 
-  pipeline.addPolicy(apiVersionPolicy(options));
+  if (options.apiVersion) {
```

Comment:
I see this pr's change is backward-compatible and just curious we need to refresh sdks before this pr, any specific reason?

https://github.com/Azure/azure-sdk-for-js/issues/29883#issuecomment-2150188495

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/29904#discussion_r1626859676 check this. in the current SDK, the apiVersion options will be passed all the way to the core, which will cause a normal scenario that really need to set the apiVersion via options.apiVersion also get this warning, in https://github.com/Azure/autorest.typescript/pull/2561#discussion_r1631453104 we have done several things
1. generate the apiVersion client options and ClientApiVersionPolicy only if it's applicable.
2. remove core-client-rest apiVersion policy in all the cases.
3. stop passing apiVersion to core.

In this way, after we refreshed all the SDKs, the only case that customer will get this warning is that they are using the previous SDKs.